### DISCUSSION TITLE
Milvus Lite, is it possible support for Ubuntu 18.04?

### DISCUSSION THREAD

**Discussion by solitude-alive** [milvus-io/milvus#36339] [Q&A and General discussion]
Since Ubuntu 18.04 is widely used, updating to 20.04 may produce unexpected products for other applications for our devices. 
Is it possible for Milvus Lite to support Ubuntu 18.04 in the future?

**Comment by yhmo** [comment]
The milvus-lite depends on milvus source code, which is compiled under Ubuntu 20.04 or later.
The soft requirement of milvus:
https://github.com/milvus-io/milvus/tree/master?tab=readme-ov-file#build-milvus-from-source-code

The soft environment of ubuntu 18.04 could not meet milvus requirement, I don't think it is possible to roll back the source code to fit it.

**Reply by solitude-alive** [reply]
Thank you for your relying.
Is it feasible to manually compile it under Ubuntu 18.04 if I make some modifications to a few dependencies? The link appears to be related to the standalone version—will that work?

**Reply by yhmo** [reply]
I think we don't intend to roll back the software environment to Ubuntu 18.04(it is risky to bring a large impact to the entire project). 

The milvus-lite includes milvus source code by git submodules, at here: https://github.com/milvus-io/milvus-lite/tree/main/thirdparty. You can fork milvus repo to create your customized milvus, and fork the milvus-lite repo, redirect the submodules to the milvus repo you forked.

**Reply by solitude-alive** [reply]
Thank you for your relying.
